provenance/repos.py

```diff
@@ -510,12 +573,12 @@ def put(self, artifact_record, read_through=False):
                                s.serializer(artifact_record))
 
             inputs_json = _inputs_json(artifact_record.inputs)
-
-            db_artifact = db.Artifact(artifact_record, inputs_json)
+            run = self._current_run
+            db_artifact = db.Artifact(artifact_record, inputs_json, run)
```

Comment:
@scizo actually, this PR totally misses the need to have the runs replicated as well! 

I like the idea of using UUIDs for the run ids, that will make things easier.